okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientTransport.java

```diff
@@ -344,6 +347,73 @@ static Status toGrpcStatus(ErrorCode code) {
     return ERROR_CODE_TO_STATUS.get(code);
   }
 
+  private Socket upgradeToTls(Socket socket, SSLSocketFactory sslSocketFactory) throws IOException {
```

Comment:
Should we just create a class (in our codebase) in the com.squareup.okhttp package in order to use ConnectionSpec.apply()? We are already using internal classes, and using ConnectionSpec seems like it would align us more with OkHttp.